ColabTool/Shared/PostDetails/index.jsx

```diff
@@ -80,94 +82,93 @@ const renderControls = ({
       <Text size={'small'}> Deleting... </Text>
     );
   }
+
+  if (manager && isWorking && !isDeleting) {
+    return (
+      <Text size={'small'}> Approving... </Text>
+    );
+  }
+
   return (
     <div>
-      {renderEdit({
-        onEditClick,
-      })}
       {renderDelete({
         isConfirmingDelete,
         onCancelConfirmClick,
         onDeleteConfirmClick,
         onDeleteClick,
       })}
+      {renderEdit({
+        onEditClick,
+      })}
+      {manager && renderApproval({
```

Comment:
I think it would be better to put the conditional logic in the `renderApproval` function here. It's not immediately obvious that `renderApproval` only get's called if `manager === true`